sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -105,7 +105,7 @@ private[sql] class CacheManager extends Logging {
     val planToCache = query.queryExecution.analyzed
     val dataIndex = cachedData.indexWhere(cd => planToCache.sameResult(cd.plan))
     require(dataIndex >= 0, s"Table $query is not cached.")
-    cachedData(dataIndex).cachedRepresentation.uncache(blocking)
+    cachedData(dataIndex).cachedRepresentation.cachedColumnBuffers.unpersist(blocking)
```

Comment:
this is slightly different with the previous version, as we also set `_cachedColumnBuffers` to null in `uncache`